### DISCUSSION TITLE
Is there a bot to automatically bump homebrew formula?

### DISCUSSION THREAD

**Discussion by furkatgofurov7** [Homebrew/discussions#4260] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

It is not a question per se, just would like to clarify if what we were thinking was correct regarding clusterctl homebrew publish on release.

**Context:** Cluster API community member updates homebrew formula of clusterctl on every release manually. 
One of the recent bump  => https://github.com/Homebrew/homebrew-core/pull/120231 was opened automatically. 
However, the following bump did not follow the same and we had to open a PR manually => https://github.com/Homebrew/homebrew-core/pull/121967

Our assumption is the maintainer is running a bot with his account, since there are a lot of PRs opened under the same github handle with similar PR descriptions:

![image](https://user-images.githubusercontent.com/40443040/220607752-e39f92c4-3be7-449f-a04a-01ee6621f8ac.png)

**Can someone please clarify:** is there a bot which opens a PR automatically so that it is taken care by it OR is it just owner of the PR run/runs a script from time to time?

**Comment by carlocab** [comment, marked as answer]
@BrewTestBot updates a list of formulae and opens PRs to update them. Others with the message that you see were opened using a `brew bump` or `brew bump-formula-pr` command. Whether this was done via a bot, a script, or manually can be answered only by the user who opened the PR.

**Reply by furkatgofurov7** [reply]
@carlocab thanks for the quick reply. I am new to the homebrew project, can you please point me to the list of formulae where bot is instructed to update automatically? maybe it is possible that `clusterctl` can be also added in that list.

**Reply by carlocab** [reply]
https://github.com/Homebrew/homebrew-core/blob/91c2e435b2f354084c9183c602081840ac9558e3/.github/workflows/autobump.yml#L15

**Reply by furkatgofurov7** [reply]
Thanks. I can see the `clusterctl` in the [list](https://github.com/Homebrew/homebrew-core/blob/91c2e435b2f354084c9183c602081840ac9558e3/.github/workflows/autobump.yml#L142), meaning bot should open a PR itself once it detects there is a new release? The last PRs it opened was some time ago https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+author%3ABrewTestBot+clusterctl+is%3Aclosed and I assume since formulae was not updated in above release series manually, bot did react and opened them, but if there is manual update before the action is run, then it disregards it.

**Reply by furkatgofurov7** [reply]
@carlocab can you confirm please ^ assumptions are correct? 

Also, one more question on the bot opened PRs: I noticed some PRs failed the publish like in https://github.com/Homebrew/homebrew-core/pull/108335#issuecomment-1219224625 so maintainers had to open the same PR manually in https://github.com/Homebrew/homebrew-core/pull/108326 (using `brew bump-formula-pr`). 
Is it so that build can fail in bot opened PRs and then we have to anyways track the bot PR to make sure it does not fail the build, and if it is the case open it manually?

**Reply by SMillerDev** [reply]
Please wait more than 30 minutes before pinging volunteers. Nobody in Homebrew gets paid enough to have a response time like that.

**Reply by SMillerDev** [reply]
> Thanks. I can see the clusterctl in the [list](https://github.com/Homebrew/homebrew-core/blob/91c2e435b2f354084c9183c602081840ac9558e3/.github/workflows/autobump.yml#L142), meaning bot should open a PR itself once it detects there is a new release?

Yes, if none are open already.

> Is it so that build can fail in bot opened PRs and then we have to anyways track the bot PR to make sure it does not fail the build, and if it is the case open it manually?

Not sure what you mean with this, but the example is a classic case of "if something fails, try again"

**Reply by furkatgofurov7** [reply]
> 
> Yes, if none are open already.
> 

> > Is it so that build can fail in bot opened PRs and then we have to anyways track the bot PR to make sure it does not fail the build, and if it is the case open it manually?
> 
> Not sure what you mean with this, but the example is a classic case of "if something fails, try again"

👍🏼 


Thank you folks for all the answers and pointers!

**Reply by p-linnane** [reply]
I opened the PR you mentioned. I periodically check for outdated formulae and run `brew bump` on my local machine to open PR's.

The `autobump` workflow runs once per day. It does not detect new formulae outside that daily execution.

**Reply by furkatgofurov7** [reply]
Thanks @p-linnane for the input.
